src/modules/auth/auth.service.ts

```diff
@@ -49,14 +101,174 @@ export default class AuthService {
     }
   }
 
-  async login(doctorDto: LoginDoctorDto): Promise<{ token: string }> {
-    const doctor = await this.validateUser(doctorDto);
-    return this.generateToken(doctor);
+  private async sendActivationMail(to: string, link: string): Promise<void> {
+    try {
+      await this.transporter.sendMail({
```

Comment:
make separate service for emails and templates, to have one place where you can change it if needed